gradle-plugin-testing-junit/src/main/java/com/palantir/gradle/testing/files/arbitrary/ArbitraryDirectory.java

```diff
@@ -18,10 +18,10 @@
 
 import com.google.errorprone.annotations.RestrictedApi;
 import com.palantir.gradle.testing.RestrictedCreation;
-import com.palantir.gradle.testing.files.FileFactory;
+import com.palantir.gradle.testing.files.Directory;
 import java.nio.file.Path;
 
-public record ArbitrarySrcDir(Path path) implements FileFactory {
+public record ArbitraryDirectory(Path path) implements Directory {
```

Comment:
I'm not sure how I feel about `ArbitrarySrcDir` being collapsed into a `ArbitraryDirectory`, feels like having the hierarchy:

```
Directory
  |
ArbitraryDirectory
  |
ArbitrarySrcDir
```

But not fully fleshed out what I would want to live on `ArbitrarySrcDir`